react/components/SearchResult.js

```diff
@@ -101,6 +110,113 @@ class SearchResult extends Component {
         showLoadingAsOverlay: true,
       })
     }
+
+    document.addEventListener('scroll', this.handleScroll)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('scroll', this.handleScroll)
+  }
+
+  handleScroll = () => {
+    const scroll = this.props.leanMode ? Infinity : window.scrollY
+    const {runtime: { hints: { mobile } }} = this.props
+    
+    if (typeof scroll !== 'number' || !mobile) return
+
+    if (scroll > this.state.scrollValue) this.handleScrollDown()
+    else if (scroll < this.state.scrollValue) this.handleScrollUp(scroll)
+
+    this.setState({scrollValue: scroll})
+  }
+
+  handleScrollUp = scrollValue => {
+    const marginValue = scrollValue * CONVERSION_MARGIN_VALUE
+
+    const searchOptionsBarElement = this.searchOptionsBar.current
+    if (searchOptionsBarElement) {
+      searchOptionsBarElement.style.transition = `opacity ${FADE_DURATION}ms ${FADE_DELAY}ms`
+      searchOptionsBarElement.style.opacity = 1
+
+      if (scrollValue < HEADER_SIZE) searchOptionsBarElement.style.marginTop = `-${marginValue}rem`
+      else searchOptionsBarElement.style.marginTop = `${MARGIN_TOP}rem`
+    }
+  }
+
+  handleScrollDown = () => {
+    const searchOptionsBarElement = this.searchOptionsBar.current
+    if (searchOptionsBarElement) {
```

Comment:
There is any cenario where `searchOptionsBarElement` is null or undefined?

Reply:
I am not sure. In certains examples I have seen this is a safe way to access the DOM element, but I think it is not a requirement. Anyway, this function is not used anymore because the strategy of appearing/disappearing has changed.